vms/avm/vm_test.go

```diff
@@ -716,3 +721,202 @@ func TestClearForceAcceptedExportTx(t *testing.T) {
+	snowCtx = snowtest.Context(t, snowtest.XChainID)
+	snowCtx.SharedMemory = atomic.NewMemory(db).NewSharedMemory(ids.ID{})
+
+	vm = &VM{
+		StateMigrationFactory: gForkStateMigrationFactory{CommitFrequency: 2},
+	}
+
+	toEngine = make(chan common.Message, 1)
+	require.NoError(vm.Initialize(
+		context.Background(),
+		snowCtx,
+		db,
+		genesisBytes,
+		nil,
+		configBytes,
+		toEngine,
+		fxs,
+		&enginetest.Sender{},
+	))
+	require.NoError(vm.SetState(context.Background(), snow.Bootstrapping))
+	require.NoError(vm.Linearize(context.Background(), ids.ID{}, toEngine))
+	require.NoError(vm.SetState(context.Background(), snow.NormalOp))
+
+	// Check that all previous state still exists
+	for itr := wantBlkIDs.NewIterator(); itr.Next(); {
+		wantBlkID := itr.Key()
+		height := itr.Value()
+
+		gotBlkID, err := vm.GetBlockIDAtHeight(context.Background(), height)
+		require.NoError(err)
+		require.Equal(wantBlkID, gotBlkID)
+
+		gotBlk, err := vm.GetBlock(context.Background(), wantBlkID)
+		require.NoError(err)
+		require.Equal(wantBlkID, gotBlk.ID())
+	}
+
+	for _, wantTxID := range wantTxs {
+		gotTx, err := vm.GetTx(wantTxID)
+		require.NoError(err)
+		require.Equal(wantTxID, gotTx.ID())
+	}
+
+	for _, wantUTXO := range wantUTXOs {
+		gotUTXO, err := vm.GetUTXO(wantUTXO.InputID())
+		require.NoError(err)
+		require.Equal(wantUTXO.InputID(), gotUTXO.InputID())
+	}
+
+	gotLastAcceptedBlkID, err := vm.LastAccepted(context.Background())
+	require.NoError(err)
+	wantLastAcceptedBlkID, _, ok := wantBlkIDs.Newest()
+	require.True(ok)
+	require.Equal(wantLastAcceptedBlkID, gotLastAcceptedBlkID)
+	require.Equal(wantLastAcceptedTimestamp, vm.GetLastAcceptedTimestamp())
+
+	db, err = memdb.Copy(db)
+	require.NoError(vm.Shutdown(context.Background()))
+
+	// Check that all previous state was deleted
+	snowCtx = snowtest.Context(t, snowtest.XChainID)
+	snowCtx.SharedMemory = atomic.NewMemory(db).NewSharedMemory(ids.ID{})
+
+	vm = &VM{StateMigrationFactory: NoStateMigrationFactory{}}
+	toEngine = make(chan common.Message, 1)
+	require.NoError(vm.Initialize(
+		context.Background(),
+		snowCtx,
+		db,
+		genesisBytes,
+		nil,
+		configBytes,
+		toEngine,
+		fxs,
+		&enginetest.Sender{},
+	))
+	require.NoError(vm.SetState(context.Background(), snow.Bootstrapping))
+	require.NoError(vm.Linearize(context.Background(), ids.ID{}, toEngine))
+	require.NoError(vm.SetState(context.Background(), snow.NormalOp))
+
+	for itr := wantBlkIDs.NewIterator(); itr.Next(); {
+		wantBlkID := itr.Key()
+		height := itr.Value()
+
+		_, err := vm.GetBlock(context.Background(), wantBlkID)
+		require.ErrorIs(err, database.ErrNotFound)
+
+		_, err = vm.GetBlockIDAtHeight(context.Background(), height)
+		require.ErrorIs(err, database.ErrNotFound)
+	}
+
+	for _, txID := range wantTxs {
+		_, err := vm.GetTx(txID)
+		require.ErrorIs(err, database.ErrNotFound)
+	}
+
+	for _, utxo := range wantUTXOs {
+		_, err := vm.GetUTXO(utxo.InputID())
+		require.ErrorIs(err, database.ErrNotFound)
+	}
+}
```

Comment:
Need to test that indexer data is also migrated correctly

Reply:
We're going to consider deprecating the indexers, so we won't have to migrate this data.